gbpservice/nfp/config_orchestrator/agent/loadbalancerv2.py

```diff
@@ -0,0 +1,457 @@
+        # lb_db = self._get_lb_context(**kwargs)
+        # db = self._filter_service_info_with_resource(lb_db, core_db)
+        db = core_db
+        return db
+
+    def _prepare_resource_context_dicts(self, **kwargs):
+        # Prepare context_dict
+        context = kwargs.get('context')
+        ctx_dict = context.to_dict()
+        # Collecting db entry required by configurator.
+        # Addind service_info to neutron context and sending
+        # dictionary format to the configurator.
+        db = self._context(**kwargs)
+        rsrc_ctx_dict = copy.deepcopy(ctx_dict)
+        rsrc_ctx_dict.update({'service_info': db})
+        return ctx_dict, rsrc_ctx_dict
+
+    def _data_wrapper(self, context, tenant_id, name, reason, nf, **kwargs):
+        nfp_context = {}
+        description = ast.literal_eval((nf['description'].split('\n'))[1])
+        if name.lower() == 'loadbalancer':
+            lb_id = kwargs['loadbalancer']['id']
+            kwargs['loadbalancer'].update({'description': str(description)})
+            nfp_context = {'network_function_id': nf['id'],
+                           'loadbalancer_id': kwargs['loadbalancer']['id']}
+        elif name.lower() == 'listener':
+            lb_id = kwargs['listener'].get('loadbalancer_id')
+            kwargs['listener']['description'] = str(description)
+        elif name.lower() == 'pool':
+            lb_id = kwargs['pool'].get('loadbalancer_id')
+            kwargs['pool']['description'] = str(description)
+        elif name.lower() == 'member':
+            pool = kwargs['member'].get('pool')
+            if pool:
+                lb_id = pool.get('loadbalancer_id')
+            kwargs['member']['description'] = str(description)
+        elif name.lower() == 'healthmonitor':
+            pool = kwargs['healthmonitor'].get('pool')
+            if pool:
+                lb_id = pool.get('loadbalancer_id')
+            kwargs['healthmonitor']['description'] = str(description)
+        else:
+            kwargs[name.lower()].update({'description': str(description)})
+            lb_id = kwargs[name.lower()].get('loadbalancer_id')
+
+        args = {'tenant_id': tenant_id,
+                'lb_id': lb_id,
+                'context': context,
+                'description': str(description)}
+
+        ctx_dict, rsrc_ctx_dict = self.\
+            _prepare_resource_context_dicts(**args)
+
+        nfp_context.update({'neutron_context': ctx_dict,
+                            'requester': 'nas_service'})
+        resource_type = 'loadbalancerv2'
+        resource = name
+        resource_data = {'neutron_context': rsrc_ctx_dict}
+        resource_data.update(**kwargs)
+        body = common.prepare_request_data(nfp_context, resource,
+                                           resource_type, resource_data,
+                                           description['service_vendor'])
+        return body
+
+    def _post(self, context, tenant_id, name, nf, **kwargs):
```

Comment:
_post and _delete methods are same as LBaaS V1. Need to generalize. Let us have a common loadbalancer.py file with all common functions for V1 and V2. We need to have specifics of V1 in loadbalancer_v1.py and V2 in loadbalaner_v2.py.